### PR TITLE
feat: stats logging

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

`stats.logging = "verbose"` =>

Screenshot 2023-08-03 at 15 49 50

## Test Plan

added in `Stats.test.ts`

### CODE REVIEW

File: packages/rspack/tests/Stats.test.ts [hunk 1]
```diff
@@ -163,7 +190,7 @@ describe("Stats", () => {
 		    entry ./fixtures/a
 		./fixtures/a.js [876] {main}
 		  entry ./fixtures/a
-		rspack compiled successfully (f0a86c7e70b0de037daf)"
+		rspack compiled successfully (1788bb8a658f2310fbf6)"
```
Comment (ahabhgk) [inline packages/rspack/tests/Stats.test.ts:193]: hash change is caused by https://github.com/web-infra-dev/rspack/pull/3893/files#diff-a3ed34d20f0b15a073bdeaac181d27e0fe74d19b63f7a0eb6cc1868192ef6dd9L66-R70, the name will be hashed in js_chunk_hash

File: crates/rspack_core/src/logger.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        self.label, start.label
+      );
+    }
+  }
+}
+
+pub type CompilationLogging = Arc<DashMap<String, Vec<LogType>, BuildHasherDefault<FxHasher>>>;
+
+pub struct CompilationLogger {
+  logging: CompilationLogging,
```
Comment (jerrykingxyz) [inline crates/rspack_core/src/logger.rs:205]: it is hard to distinguishing between `CompilationLogging` and `CompilationLogger`, maybe we can change the name.

File: crates/rspack_plugin_devtool/src/lib.rs [hunk 3]
```diff
@@ -100,6 +102,8 @@ impl Plugin for DevtoolPlugin {
     _ctx: PluginContext,
     args: ProcessAssetsArgs<'_>,
   ) -> PluginProcessAssetsOutput {
+    let logger = args.compilation.get_logger(self.name());
```
Comment (jerrykingxyz) [inline crates/rspack_plugin_devtool/src/lib.rs:105]: Requiring a compilation for `logger` generation doesn not seem like a good idea. maybe we can
```
let logger = Logger::new(self.name());
....
self.compilation.addLogger(logger);
// or
self.compilation.loggers.push(logger);
```

File: crates/node_binding/src/js_values/stats.rs [hunk 4]
```diff
@@ -377,6 +469,16 @@ impl JsStats {
       .collect()
   }
 
+  #[napi]
+  pub fn get_logging(&self) -> Vec<JsStatsLogging> {
```
Comment (jerrykingxyz) [inline crates/node_binding/src/js_values/stats.rs:473]: Logging is hard to distinguish from "compilationLogging", maybe we could name something more specific.
```
pub fn get_js_stats_logging(&self) -> Vec<JsStatsLogging>;
// or
pub fn get_stats_logging(&self) -> Vec<JsStatsLogging>;
```

File: crates/rspack_plugin_devtool/src/lib.rs (hunk 3 above)
Comment (ahabhgk) [reply crates/rspack_plugin_devtool/src/lib.rs:105]: `compilation.addLogger` also require a compilation, `compilation.get_logger` API style is aligned with webpack

File: crates/rspack_core/src/logger.rs (hunk 2 above)
Comment (ahabhgk) [reply crates/rspack_core/src/logger.rs:205]: logging and logger, this is how webpack naming them...🥲

File: crates/node_binding/src/js_values/stats.rs (hunk 4 above)
Comment (ahabhgk) [reply crates/node_binding/src/js_values/stats.rs:473]: It's easy to distinguish them by the function signature, and logging basically only consumed by stats, so specific `get_logging` to `get_stats_logging` seems not necessary

**Review by jerrykingxyz — APPROVED** [review]

**Pull request merged**